src/server/accounting.c

```diff
@@ -2259,3 +2259,86 @@ void log_alter_records_for_attrs(job *pjob, svrattrl *plist) {
 	if (entire_record[0] != '\0')
 		account_record(PBS_ACCT_ALTER, pjob, entire_record);
 }
+
+
+/**
+ * @brief
+ * Common function to log a suspend/resume record 
+ * for suspend/resume job events respectively.
+ * 
+ * @param[in] pjob - job to log records for.
+ * @param[in] user - User who requested the signal suspend/resume 
+ * @param[in] host - Hostname where the signal was triggered
+ * @param[in] signal_type - flag indicating the type of event suspend/resume
+ * 
+ * @returns void 
+ */
+void
+log_suspend_resume_record(job *pjob, char *user, char *host, char *signal_type)
+{
+	char *log_record = NULL;
+	char *tmp;
+	struct svrattrl *patlist = NULL;
+	char *resc_used;
+	int resc_used_size = 0;
+	pbs_list_head temp_head;
+	int acct_type = PBS_ACCT_RESUME;
+	int rc;
+
+	rc = pbs_asprintf(&log_record, "requestor=%s@%s", user, host);
+
+	if (strcmp(signal_type, SIG_SUSPEND)==0 || strcmp(signal_type, SIG_ADMIN_SUSPEND) == 0) {
+		acct_type = PBS_ACCT_SUSPEND;
+		
+		if (pjob->ji_wattr[JOB_ATR_resc_released].at_flags & ATR_VFLAG_SET) {
+			tmp = log_record;
+			rc = pbs_asprintf(&log_record, "%s resources_released=%s", tmp, 
+			pjob->ji_wattr[JOB_ATR_resc_released].at_val.at_str);
+			free(tmp);
+		}	
+
+		CLEAR_HEAD(temp_head);
```

Comment:
Since most of the following code for creating resources_used string for accounting is common with the E record construction, put it in a separate function to avoid code duplication?